clang/test/CXX/temp/temp.decls/temp.friend/p1.cpp

```diff
@@ -373,6 +373,7 @@ template <class T> void foo() {} // expected-note{{candidate ignored: not a memb
 }
 using ns::foo;
 template <class T> struct A {
+  // expected-error@+1{{friend function specialization cannot be defined}}
```

Comment:
These already error, and doing a double-error here is not the right way to go.  In this case, it seems to make sense that the 2nd error doesn't happen, same with the next one.

Reply:
Per my other comment, I think the right thing to do here is to diagnose friend function definitions after calling `ActOnFunctionDecl`... in this case we would emit the "no candidate function template was found for dependent friend function template specialization" diagnostic, but not "friend function specialization cannot be defined".